src/models.py

```diff
@@ -931,6 +959,45 @@ def _lm_forward(self, batch, task, predict):
             pass
         return out
 
+    def _lm_only_lr_forward(self, batch, task, predict):
+        """Only left to right pass for LM model - non-bidirectional models.
+        Args:
+            batch: indexed input data
+            task: (Task obejct)
+            predict: (boolean) predict mode (not supported)
+        return:
+            out: (dict)
+                - 'logits': output layer, dimension: [batchSize * timeSteps * 2, outputDim]
+                            first half: [:batchSize*timeSteps, outputDim] is output layer from forward layer
+                            second half: [batchSize*timeSteps:, outputDim] is output layer from backward layer
+                - 'loss': size average CE loss
+        """
+
+        out = {}
+        sent_encoder = self.sent_encoder
+        assert_for_log('targs' in batch and 'words' in batch['targs'],
+                       "Batch missing target words!")
+        pad_idx = self.vocab.get_token_index(self.vocab._padding_token, 'tokens')
+        b_size, seq_len = batch['targs']['words'].size()
+        n_pad = batch['targs']['words'].eq(pad_idx).sum().item()
+        out['n_exs'] = (b_size * seq_len - n_pad) * 2
+        sent, mask = sent_encoder(batch['input'], task)
+        sent = sent.masked_fill(1 - mask.byte(), 0)
+        hid2voc = getattr(self, "%s_hid2voc" % task.name)
+        # left to right LM logits
+        logits = hid2voc(sent).view(b_size * seq_len, -1)
+        out['logits'] = logits
+        trg_fwd = batch['targs']['words'].view(-1)
+        # set targs to contain only left to right LM targets.
+        targs = trg_fwd
+        assert logits.size(0) == targs.size(0), "Number of logits and targets differ!"
+        # cross entropy for loss
+        out['loss'] = F.cross_entropy(logits, targs, ignore_index=pad_idx)
+        task.scorer1(out['loss'].item())
+        if predict:
```

Comment:
This doesn't do anything. Why is it here?

Reply:
Removed